### ISSUE TITLE
[REQUEST] Customize setting the `resources` on the rbac-proxy container

### ISSUE THREAD

**Issue by DeviaVir** [bugfest/tor-controller#82]
**Is your feature request related to a problem? Please describe.**

It looks like because we are not currently setting any resource requests/limits, some (looking at you GKE) schedulers come up with crazy (expensive) numbers like:
```
        resources:
          limits:
            cpu: 500m
            ephemeral-storage: 1Gi
            memory: 2Gi
          requests:
            cpu: 500m
            ephemeral-storage: 1Gi
            memory: 2Gi
```

**Describe the solution you'd like**

Since we want to stay away of setting defaults, let's at least make it possible to customize these values.

**Describe alternatives you've considered**

Considered investigating if [autopilot.gke.io/primary-container](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-resource-requests#init-containers) could help with this, but it already says it applies to the first container (which is the tor-controller already)

**Additional context**

![Screenshot 2024-12-30 17 58 46](https://github.com/user-attachments/assets/4aa41590-8ab8-4057-b839-fb92e05224b8)

**Comment by bugfest** [comment]
Fixed by #83

**Comment by DeviaVir** [comment]
![Screenshot 2024-12-31 10 46 41](https://github.com/user-attachments/assets/a8df28f3-d973-450b-b5e2-056bc43cd257)
Glorious. Thank you for the swift turnaround!
